cartographer/mapping/sparse_pose_graph.cc

```diff
@@ -35,6 +35,40 @@ proto::SparsePoseGraph::Constraint::Tag ToProto(
   LOG(FATAL) << "Unsupported tag.";
 }
 
+SparsePoseGraph::Constraint::Tag FromProto(
+    const proto::SparsePoseGraph::Constraint::Tag& proto) {
+  switch (proto) {
+    case proto::SparsePoseGraph::Constraint::INTRA_SUBMAP:
+      return SparsePoseGraph::Constraint::Tag::INTRA_SUBMAP;
+    case proto::SparsePoseGraph::Constraint::INTER_SUBMAP:
+      return SparsePoseGraph::Constraint::Tag::INTER_SUBMAP;
+  }
+  LOG(FATAL) << "Unsupported tag.";
+}
+
+std::vector<SparsePoseGraph::Constraint> FromProto(
+    const ::google::protobuf::RepeatedPtrField<
+        ::cartographer::mapping::proto::SparsePoseGraph::Constraint>&
+        constraint_protos) {
+  std::vector<SparsePoseGraph::Constraint> constraints;
+  for (const auto& constraint_proto : constraint_protos) {
+    const mapping::SubmapId submap_id{
```

Comment:
Perhaps we should `CHECK` components of `node_id` and `submap_id` for being `>= 0` here.

Reply:
MapById already checks that, so let's not add redundant checks.